service/matching/taskListManager_test.go

```diff
@@ -219,68 +218,78 @@ func TestDescribeTaskList(t *testing.T) {
 	require.Zero(t, taskListStatus.GetBacklogCountHint())
 }
 
-func tlMgrStartWithoutNotifyEvent(tlm *taskListManagerImpl) {
-	// mimic tlm.Start() but avoid calling notifyEvent
-	tlm.liveness.Start()
-	tlm.startWG.Done()
-	go tlm.taskReader.dispatchBufferedTasks(defaultTaskBufferIsolationGroup)
```

Comment:
sorry, small brain following along here, are we ensuring these daemons are started in the new test?

Reply:
I looked at the code - these daemons are starting among with tlm.taskReader.Start():
```
for g := range tr.taskBuffers {
	go tr.dispatchBufferedTasks(g)
}
go tr.getTasksPump()
```

I think the previous approach where we started *selected* things bypassing the tlm methods actually a bad approach. We were trying to avoid notifyEvent (which is, I believe, no longer happening in .Start), but also missed some important changes like starting the writer.

I think it is really important for us to cover the happy-paths as well. Otherwise tests like `TestAddTaskStandby` pass unintentionally: it assumed we had to stop the writer though it's never been started!